elastic/kibana
# [Fleet] Improve support for Agentless in Fleet UI

Closes https://github.com/elastic/kibana/issues/180375

## Summary
Expand support for Agentless in Fleet UI. Implemented on this PR:
- Agent policies created with `supports_agentless` (added in https://github.com/elastic/kibana/pull/182709) are now marked as `is_managed` 
- Added support for `deployment_modes.agentless.enabled` and `hide_in_deployment_modes` introduced with https://github.com/elastic/package-spec/issues/684:
  - `hide_in_deployment_modes` shows/hides variables in agentless/default UI
  - `deployment_modes.agentless.enabled: true` determines an integration as enabled for agentless

## Notes
- In this PR, an integration enabled for serverless is still added to the default `agentless` policy. 
- Integrations enabled for agentless don't display any specific "agentless" form yet. That part is TBD

### Testing
- Run local env for [agentless](https://docs.elastic.dev/security-solution/cloud-security/agentless)

### Verify that new settings are honoured in integration policy
- Use the compiled package with new properties defined in https://github.com/elastic/package-spec/pull/738 `agentless_test_package-1.0.1-rc1.zip` (Packages in this [comment](https://github.com/elastic/kibana/pull/183045#issuecomment-2124213301))
- Upload it to Kibana:
  ```sh
   curl -k -XPOST -H 'content-type: application/zip' -H 'kbn-xsrf: true' https://localhost:5601/api/fleet/epm/packages -u elastic_serverless:changeme --data-binary @agentless_test_package-0.0.1.zip
  ```
- Check that the variable marked with `hide_in_deployment_modes: "agentless"` are not visible
- Check that the agent policies selector is not visible

![Screenshot 2024-05-22 at 10 37 42](https://github.com/elastic/kibana/assets/16084106/02f12ebf-f448-4cfa-b9e7-8cb1a1226e5e)


### Verify that agent policies with `supports_agentless` are marked as managed
- Create a new policy with `supports_agentless`
  ```
  POST kbn:/api/fleet/agent_policies
  {
    "name": "Test - Agentless policy",
    "namespace": "default",
    "supports_agentless": true
  }
  ```
- Check that the created policy has `is_managed = true`
- Try to change the value to false - an error is thrown
  ```
  PUT kbn:/api/fleet/agent_policies/<policy_id>
  {
    "name": "Test - Agentless policy",
    "namespace": "default",
    "supports_agentless": false
  }
  ```


### Checklist
